lms/djangoapps/instructor/lti_grader.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+"""
+Send grades to an OpenEdX LTI component
+"""
+import csv
+
+from instructor import lti_connection
+from instructor.lti_connection import LTIConnectionError
+
+
+class LTIGrader(object):
+    """
+    Updates grades for a given LTI component
+    """
+
+    def __init__(self, course_id, url_base, lti_key, lti_secret):
+        self.url_base = url_base
+        self.course_id = course_id
+        self.key = lti_key
+        self.secret = lti_secret
+        self.output = {'success': '', 'error': ''}
+
+    def _get_first_anon_id(self, grade_data):
+        """
+        Returns the first anonymized student id for in the given uploaded csv data file
+        :param grade_data: an InMemoryUploadedFile with lti grade data
+        :return: anon_id
+        """
+        reader = self._unicode_csv_reader(grade_data)
+        try:
+            reader.next()
+            first_row = reader.next()
+            anon_id = first_row[1]
+        except StopIteration:
+            self.output['error'] += '\nThe CSV file contains no useful data!'
+            anon_id = None
+        return anon_id
+
+    def update_grades(self, grade_data):
+        """
+        Updates grades for the LTI component with the grade data provided
+        :param grade_data: an InMemoryUploadedFile with lti grade data
+        :return: output, a dictionary with two keys: 'success' and 'error'
+        """
+        grade_data = grade_data.read().splitlines()
+
+        first_anon_id = self._get_first_anon_id(grade_data)
+        if first_anon_id is None:
+            return self.output
+
+        test_url = self.url_base + first_anon_id
+        try:
+            lti_connection.validate_lti_passport(self.key, self.secret, test_url)
+        except LTIConnectionError as error:
+            self.output['error'] += '\n' + unicode(error)
```

Comment:
I don't care for all of this string concatenation for error messages, especially prepending each with a literal newline (_especially_ w/in the body of loops).

Could we compile them as an array and then join the messages when needed?

```python
self.errors = []
# ...
self.errors.append(u'Uh-oh!')
self.errors.append(u'Breaking stuff')
# ...
self.output['error'] = u'\n'.join(self.errors)
```

Or, if we anticipate multiple messages, we could even return the list of errors as JSON and leave that up to the front end to display them all..

The latter approach sounds like cleanest.